src/acl/ParameterizedNode.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_ACL_PARAMETERIZEDNODE_H
+#define SQUID_SRC_ACL_PARAMETERIZEDNODE_H
+
+#include "acl/Acl.h"
+#include "acl/Data.h"
+
+#include <memory>
+
+namespace Acl
+{
+
+/// An ACL that manages configured parameters using a given Parameters class P.
+/// That Parameters class must support an ACLData<> or equivalent API.
+template <class P>
+class ParameterizedNode: public ACL
+{
+public:
+    using Parameters = P;
+
+    // to avoid dragging constructor parameters through each derived class, they
+    // are set in a leaf class constructor; \sa Acl::FinalizedParameterizedNode
+    ParameterizedNode() = default;
+    ~ParameterizedNode() override = default;
+
+protected:
+    /* ACL API */
+    void prepareForUse() override { parameters->prepareForUse(); }
+    void parse() override { parameters->parse(); }
+    SBufList dump() const override { return parameters->dump(); }
+    bool empty() const override { return parameters->empty(); }
+    const Acl::Options &lineOptions() override { return parameters->lineOptions(); }
+
+    /// items this ACL is configured to match; never nil after construction
```

Comment:
However, as you noted above, it is 'never nil' after a derived class construction (not this class construction). Also, can we  somehow guarantee (during compile time) that this derived class initialization occurred?